rocket/src/main/scala/ipxact/AXI4GCD.scala

```diff
@@ -0,0 +1,109 @@
+// See LICENSE for license details.
+
+package ipxact
+
+import chisel3._
+import chisel3.iotesters.PeekPokeTester
+import freechips.rocketchip.amba.axi4._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.util._
+import freechips.rocketchip.regmapper.{RegField, RegFieldDesc, RegReadFn, RegWriteFn}
+
+case class GCDParams(
+  width: Int = 32
+)
+
+class GCD(params: GCDParams = GCDParams()) extends Module {
+  val io = IO(new Bundle {
+    val a  = Input(UInt(params.width.W))
+    val b  = Input(UInt(params.width.W))
+    val e  = Input(Bool())
+    val z  = Output(UInt(params.width.W))
+    val v  = Output(Bool())
+  })
+
+  Annotated.params(this, params)
+
+  val x = Reg(UInt(32.W))
+  val y = Reg(UInt(32.W))
+  when (x > y)   { x := x -% y }
+    .otherwise     { y := y -% x }
+  when (io.e) { x := io.a; y := io.b }
+  io.z := x
+  io.v := y === 0.U
+}
+
+class AXI4GCD extends LazyModule()(Parameters.empty) {
+
+  val regs = AXI4RegisterNode(AddressSet(0x0, 0xFFFF), beatBytes = 4, concurrency = 1)
+
+  val ioMemNode = BundleBridgeSource(() => AXI4Bundle(AXI4BundleParameters(addrBits = 8, dataBits = 32, idBits = 1)))
+
+  regs :=
+    BundleBridgeToAXI4(AXI4MasterPortParameters(Seq(AXI4MasterParameters("bundleBridgeToAXI4")))) :=
+    ioMemNode
+
+  val ioMem = InModuleBody { ioMemNode.makeIO() }
+
+
+
+  lazy val module = new LazyModuleImp(this) {
+    Annotated.addressMapping(ioMem,
```

Comment:
Maybe all IPXact should be added via Aspects, b/c I think this is usually how it should be done in practice.